### ISSUE TITLE
Why does SILE's \href display large red rectangles around URL targets?

### ISSUE THREAD

**Issue by ctrlcctrlv** [sile-typesetter/sile#1201]
![image](https://user-images.githubusercontent.com/838783/131994143-83115452-a569-42eb-8c64-c5001623e102.png)

(PDF viewer in image above is [Okular](https://okular.kde.org/).)

**Comment by vlasakm** [comment]
[SILE just inserts a PDF annotation that sets the border color to red](https://github.com/sile-typesetter/sile/blob/v0.11.0/packages/pdf.lua#L96). [The default border width is 1 point](https://www.adobe.com/content/dam/acom/en/devnet/pdf/pdfs/PDF32000_2008.pdf#G11.2108375).

This should probably be customizible, but I don't know SILE enough to tell how.

**Comment by ctrlcctrlv** [comment]
That's a crazy spec default, but online searching tells me that it's customizable by the writer. SILE ought to do so.

**Comment by vlasakm** [comment]
The PDF library just expects a raw PDF string for the annotation, so any customization is possible for the end user, if SILE allows it with something higher level.

I also think that setting the border to zero points (to disable it) is a saner default, unless the user requests colored borders.

For reference this is the full PDF code for such type of annotations:
```
<<
  /Type /Annot
  /Border [ 0 0 0 ]
  /Rect [ 90.791 758.072 174.438 770.027 ]
  /Subtype /Link
  /A <<
    /S /URI
    /URI (https://github.com)
  >>
>>
```

`/Rect` (which specifies the active page area) is handled  specially, the rest of the string is constructred by SILE.

**Comment by alerque** [comment]
Okular has a setting for highlighting links, but these are showing up with that *off*! I see them in Okular and Evince but not Zathura (my daily driver):

![image](https://user-images.githubusercontent.com/173595/131997687-5a805886-45c4-49bd-a9d3-d43cecc07883.png)

**Comment by alerque** [comment]
Thanks for the info @vlasakm!

Yes I agree this should be disabled by default unless requested by the user.

**Comment by vlasakm** [comment]
@alerque Since the annotation style (border and its color or something more complicated) is set by the document author I would expect all viewers to respect it.

But it is also my experience that Zathura and MuPDF (upon which Zathura is usually based) ignore the borders. I personally find them hideous and as a fellow Zathura user enjoy their absence.

In Okular I could find "Draw border around links" option under "Accesibility", and it seems to just draw an independent border (so there are two of them on top of each other if document has them).

There has to be some default and maybe the PDF one is fine. LaTeX's hyperref package does default to bordered links (with different colors for different link types). ConTeXt defaults to no borders (requires command line option to get them) and instead distinguishes them by _text_ color, but also suggests not using any color for links:

> A neutral setup makes sense because nowadays the reader kind of knows what can be clicked on.

**Comment by Omikhleia** [comment]
Just tested it works (= no border) if one replaces that line https://github.com/sile-typesetter/sile/blob/v0.11.0/packages/pdf.lua#L94 :
```
      local d = "<</Type/Annot/Subtype/Link/C [ 1 0 0 ]/A<<" .. target .. "(" .. dest .. ")>>>>"
```
With
```
      local d = "<</Type/Annot/Subtype/Link/Border [ 0 0 0 ]/A<<" .. target .. "(" .. dest .. ")>>>>"
```
The `/C [r g b]` is the border color, likely not really needed if border is disabled (defaults may vary, on my viewer it's black). On the other end it should be customizable too.

The `/B` sets the border, by default `[ 0 0 1 ]` - the first numbers are for rounded boxes, the last is the line width (and extra numbers can set dashes too). 

(For the record, PDF 32000-1:2008 Table 164. Maybe some other options from §12.5.4 can be tested too.)

Having document settings for this at least for two important options here (border color and border width), as well as per-command option, should be easy. If you are not in a hurry, I volunteer.

**Comment by Omikhleia** [comment]
For safekeeping: Instead of /Border, one may also use /BS (border style definition - as per PDF 1.7 §12.5.4 and I have to admin, looking at LaTeX's hyperref package to understand how to use it)... and get interesting options, e.g.
```
   local d = "<</Type/Annot/Subtype/Link/BS<</Type/Border/S/U/W 0.6>>/C [0 0 1]/A<<" .. target .. "(" .. dest .. ")>>>>"
```
Giving:
![image](https://user-images.githubusercontent.com/18075640/132072908-59412908-61f0-458d-ac0b-c90f41f9036b.png)

The line is a bit close to the text, but it's easy to fix (e.g. sub-stracting some padding to ``lly``

**Comment by alerque** [comment]
I suggest we setup a callback function that outputs various formatting from easy to weak inputs, and then add a package option (as a class option?) that can be passed the callback. The `\href` function might also be extended to accept the same callback on a case by case override, but the package default is probably the more important feature to add.

**Comment by alerque** [comment]
(Alternatively and possible easier to implement we can just have it be function options and then use then have users use `SILE.setCommandDefaults()`)

**Comment by alerque** [comment]
Addressed in #1209.
